src/deterministic.jl

```diff
@@ -0,0 +1,130 @@
+"""
+Construct Deterministic from a Dict of TimeArrays, DataFrames or Arrays.
+
+# Arguments
+- `name::AbstractString`: user-defined name
+- `data::Union{Dict{Dates.DateTime, Any}, SortedDict.Dict{Dates.DateTime, Any}}`: time series data. The values in the dictionary should be TimeSeries.TimeArray or be able to be converted
+- `normalization_factor::NormalizationFactor = 1.0`: optional normalization factor to apply
+  to each data entry
+- `scaling_factor_multiplier::Union{Nothing, Function} = nothing`: If the data are scaling
+  factors then this function will be called on the component and applied to the data when
+  [`get_time_series_array`](@ref) is called.
+- `timestamp = :timestamp`: If the values are DataFrames is passed then this must be the column name that
+  contains timestamps.
+- `resolution = nothing : If the values are a Matrix or a Vector, then this must be the resolution of the forecast in Dates.Period`
+"""
+function Deterministic(
+    name::AbstractString,
+    data::Union{Dict{Dates.DateTime, Any}, SortedDict{Dates.DateTime, Any}};
+    normalization_factor::NormalizationFactor = 1.0,
+    scaling_factor_multiplier::Union{Nothing, Function} = nothing,
+    timestamp = :timestamp,
+    resolution::Union{Dates.Period, Nothing} = nothing,
+)
+    for (k, v) in data
+        if v isa DataFrames.DataFrame
+            data[k] = TimeSeries.TimeArray(v; timestamp = timestamp)
+        elseif v isa TimeSeries.TimeArray
+            continue
+        else
+            try
+                data[k] =
+                    TimeSeries.TimeArray(range(k, length = length(v), step = resolution))
+            catch e
+                throw(ArgumentError("The values in the data dict can't be converted to TimeArrays. Resulting error: $e"))
+            end
+        end
+    end
+
+    ta = handle_normalization_factor(ta, normalization_factor)
+    return Deterministic(name, ta, scaling_factor_multiplier)
+end
+
+# TODO: need to make concatenation constructors for Probabilistic
```

Comment:
Is it valid? yes -> Move. No -> Remove ?

Reply:
This was a long-standing TODO.